docs/nebula-design-concepts.md

```diff
@@ -0,0 +1,216 @@
+one tag, and Nebula allows you to set multiple tags per vertex.
+
+### Edgetype
+
+Each edge has an edgetype, and each edgetype defines the properties of the edge.
+这是按照我的理解方式写的
+-->
+
+
+### Vertex, type of vertex (tag) and the corresponding vertex properties
+
+In Nebula Graph, the type of vertex is called **tag**. **A vertex must have at least one type (tag)** or multiple types (tag). **Each tag has a corresponding set of properties called schema.**
+
+Consider the above picture as example, there are two types of vertices: **player** and **team**. Vertex player's schema has three properties: id (vid), Name(string), and Age(int); vertex team's schema has two properties: id (vid) and Name(string).
+
+Like Mysql, Nebula Graph is a strong schema database. The property name and data type are determined before the data is written.
+
+### Edge, edgetype and the corresponding edge properties
+
+**Edges in Nebula Graph always have a type, a start (src)and an end vertex (dst), and a direction.** They can be self-referencing/looping but can never be dangling (missing a start or end node). **Each edge has an edgetype, and each edgetype defines the properties of the edge.**
+
+In the preceding example, there are two kinds of edges, one is **like** pointing
+from player to player, the property is likeness (double); the other is **serve**,
+pointing from team to serve, the properties are start_year(int) and end_year(int).
+
+It should be noted that there may be **multiple edges of the same or different types
+at the same time** between the start (src)and end vertex (dst).
+
+
+
+## Graph partition
+
+![image][example2]
+
+Since the number of vertices in a super large-scale relational network can reach
+to dozen billions or trillions, and the number of edges can even be dozen trillions,
+the storage of both will exceed the capacity of a general server. Therefore, a
+method is needed to cut the graph elements and store them on different logical partitions. Nebula Graph conducts **edge splitting** to make partitions, and its default strategy is Hash. The number of partitions is static and cannot be changed.
+
+
+## Data model
+
+![image][example3]
+
+In Nebula Graph, each vertex is modeled as key-values, then stored to the corresponding partition after hashing according to its vertexID.
+
+An edge is modeled as two separate key-values in Nebula Graph, i.e. out-key and
+in-key, respectively. The out-key and the edge's corresponding starting point are stored in one partition, while the in-key and end point are stored in another one.
+
+![image][example4]
+
+## Architecture
+
+Nebula Graph includes four modules: storage layer, metadata service, computation layer
+and client console.
+
+![image][example5]
+
+### Storage layer
+
+The corresponding process of storage layer is nebula-storaged, and its core is the
+distributed Key-value Storage based on RAFT protocol. Currently it supports
+multiple storage engines like Rocksdb and HBase, etc.
```

Comment:
Storage Service provides a key-value store. Separation of storage and compute makes this layer flexible, and multiple storage engines are supported. RocksDB is the default storage engine. To build a resilient distributed system, raft is implemented as the consensus algorithm.